dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetVersion.java

```diff
@@ -126,7 +126,7 @@ public int compare(DatasetVersion o1, DatasetVersion o2) {
     @ManyToOne
     private Dataset dataset;
 
-    @OneToMany(mappedBy = "datasetVersion", cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST})
+    @OneToMany(mappedBy = "datasetVersion", cascade = {CascadeType.REMOVE, CascadeType.MERGE, CascadeType.PERSIST}, orphanRemoval = true)
```

Comment:
Has this been thought thru? Do we really want to **always** remove all `fileMetadatas` upon `DatasetVersion` removal?

Reply:
This is not how it works, if we remove filemetadata from DatasetVersion List, then the filemetadata on the "other side" will be removed (since filemetadata is uniqe to each DatasetVersion there is no reason to keep it). Otherwise without this property you would have to manually remove it from both sides.

Reply:
You're right. 
"orphanRemoval is an entirely ORM-specific thing. It marks "child" entity to be removed when it's no longer referenced from the "parent" entity"

Reply:
But I think it is good topic for daily, maybe I missed something.